target/linux/ramips/dts/mt7628an_netgear_r6080.dts

```diff
@@ -0,0 +1,147 @@
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "netgear,r6080", "mediatek,mt7628an-soc";
+	model = "Netgear R6080";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		lan {
+			label = "r6080:green:lan";
+			gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power: power {
+			label = "r6080:green:power";
+			gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "r6080:green:wlan2g";
+			gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan_orange {
+			label = "r6080:orange:wlan2g";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "r6080:green:wan";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+		};
+
+		wan_orange {
+			label = "r6080:orange:wan";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+};
+
+&state_default {
+	gpio {
+		ralink,group = "p0led_an", "p1led_an", "p2led_an",
+			       "p3led_an", "p4led_an", "wdt",
+			       "wled_an", "uart1";
+		ralink,function = "gpio";
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Does this support higher speed?

Reply:
The data sheet for the chip doesn't give a definitive answer (different values for different scenarios, some of which work and some of which do not). Trial-and-error suggests that it works up to a max value of 64000000.

Reply:
Use `m25p,fast-read;` property to get a higher frequency such as 80000000

Reply:
Funny you should suggest that, am playing with that right now. That lets it support up to 80000000, but anecdotally, the initial boot with FAST_READ enabled at 80 MHz takes almost twice as long as when its only set to 64 MHz. Should I try for the highest max freq regardless of time, or try to optimize performance?

Reply:
Initial boot with FAST_READ enabled at 80 MHz - ~140 seconds
Initial boot with FAST_READ enabled at 64 MHz - ~64 seconds
Initial boot with FAST_READ disabled at 64 MHz - ~78 seconds

Reply:
Have you actually tested read speed?
`time cat /dev/mtd3 > /dev/null`

Reply:
Ah, that definitely shows that higher is better. The data sheet specifies up to 86 MHz, and that works. Thanks!